app/server.js

```diff
@@ -7,7 +7,147 @@ var bodyParser   = require('body-parser');
 var uid = require('rand-token').uid;
 
 const PORT = 3001;
+var router = Router();
+router.use(bodyParser.json());
+router.use(bodyParser.urlencoded({ extended: true }));
 
-http.createServer(function (request, response) {
+// Array of users, States, e.g, authentication, cart
+// {
+//   cart: [],
+//   login: {
+//     username: 'philip',
+//     password: 'loggedin'
+//   }
+// }
+let users = [];
+let brands = [];
+let products = [];
+let loggedInUser = undefined;
 
-}).listen(PORT);
\ No newline at end of file
+module.exports=http.createServer(function (request, response) {
+  router(request, response, finalHandler(request, response));
+}).listen(PORT, () => {
+  users = JSON.parse(fs.readFileSync("initial-data/users.json","utf-8"));
+  loggedInUser = users[0];
+
+  // Read the brands file
+  brands = JSON.parse(fs.readFileSync("initial-data/brands.json", "utf-8"));
+  //  Read the products file
+  products = JSON.parse(fs.readFileSync("initial-data/products.json", "utf-8"));
+
+});
+
+// Frontend
+// localhost: 3000 -> UI -> get me all the brands from API -> localhost:3001/api/brands -> render
+// fetch( like in "openweathermap.com/....")
+// API
+// localhost: 3001 -> API
+
+// localhost:3001/api/brands
+router.get('/api/brands', function (req, res) {
+  res.setHeader('Content-Type', 'application/json');
+  res.end(JSON.stringify(brands));
+});
+
+// Get all the products matching the brand id = categoryId
+router.get('/api/brands/:id/products', function (req, res) {
+  const { id } = req.params;
+
+  const brandProducts = [];
+
+  for (let i = 0; i < products.length; i++) {
+    if (products[i].categoryId === id) {
+      brandProducts.push(products[i]);
+    }
+  }
+  if (brandProducts.length === 0) {
+  res.writeHead(404, { "Content-Type": "application/json"});
+  } else {
+    res.writeHead(200, { "Content-Type": "application/json"});
+  }
+  res.end(JSON.stringify(brandProducts));
+});
+
+//GET products
+router.get('/api/products', function (req, res) {
+  console.log('get the products');
+  res.setHeader('Content-Type', 'application/json');
+  res.end(JSON.stringify(products))
+});
+
+//POST login, still need to mock data.
+router.post('/api/login', function (req, res) {
+  console.log('user login');
+  if (req.body.username && req.body.password) {
+    loggedInUser = users.find(user => {
```

Comment:
You never return `loggedInUser` - which is what one would expect from this route. We also would expect a token.